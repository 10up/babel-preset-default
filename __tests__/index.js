const path = require('path');
const fs = require('fs');
const babel = require('@babel/core');
const babelPresetDefault = require('../index');

const nodeModulesPath = path.join(__dirname, '../node_modules');
const reactPath = path.join(nodeModulesPath, 'react');

const es6Input = fs.readFileSync(path.join(__dirname, '../fixtures/es6+.js'));
const reactInput = fs.readFileSync(path.join(__dirname, '../fixtures/react.js'));

describe('Babel preset default', () => {
	test('transpiles ES6+ code properly', () => {
		const output = babel.transform(es6Input, {
			configFile: false,
			envName: 'production',
			presets: [babelPresetDefault],
		});

		expect(output.code).toMatchSnapshot();
	});

	test('transpiles ES6+ code properly with runtimeESModules set to true', () => {
		const output = babel.transform(es6Input, {
			configFile: false,
			envName: 'production',
			presets: [[babelPresetDefault, { runtimeESModules: true }]],
		});

		expect(output.code).toMatchSnapshot();
	});

	test('transpiles react code properly', () => {
		const output = babel.transform(reactInput, {
			configFile: false,
			envName: 'production',
			presets: [babelPresetDefault],
		});

		expect(output.code).toMatchSnapshot();
	});

	test('transpiles wordpress code properly', () => {
		const output = babel.transform(reactInput, {
			configFile: false,
			envName: 'production',
			presets: [[babelPresetDefault, { wordpress: true }]],
		});

		expect(output.code).toMatchSnapshot();
	});

	// Babel will automatically set modules to false when called by Webpack.
	test('transpiles without transforming es6 imports', () => {
		const output = babel.transform(es6Input, {
			configFile: false,
			envName: 'production',
			presets: [[babelPresetDefault, { modules: false }]],
		});

		expect(output.code).toMatchSnapshot();
	});

	test('transpiles without removing proptypes in dev mode', () => {
		const output = babel.transform(reactInput, {
			configFile: false,
			envName: 'development',
			presets: [[babelPresetDefault]],
		});

		expect(output.code).toMatchSnapshot();
	});
});

describe('Babel preset default with automatic runtime', () => {
	beforeEach(() => {
		fs.writeFileSync(path.join(reactPath, 'jsx-runtime.js'), 'export default () => {};');
	});

	afterEach(() => {
		fs.unlinkSync(path.join(reactPath, 'jsx-runtime.js'));
	});

	test('transpiles react code properly with automatic runtime', () => {
		const output = babel.transform(reactInput, {
			configFile: false,
			envName: 'production',
			presets: [babelPresetDefault],
		});

		expect(output.code).toMatchSnapshot();
	});

	test('transpiles wordpress code properly even when jsx-runtime.js is present', () => {
		const output = babel.transform(reactInput, {
			configFile: false,
			envName: 'production',
			presets: [[babelPresetDefault, { wordpress: true }]],
		});

		expect(output.code).toMatchSnapshot();
	});
});
