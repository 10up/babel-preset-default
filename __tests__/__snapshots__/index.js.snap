// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Babel preset default transpiles ES6+ code properly 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\");

require(\\"core-js/modules/es.symbol\\");

require(\\"core-js/modules/es.symbol.description\\");

require(\\"core-js/modules/es.symbol.iterator\\");

require(\\"core-js/modules/es.array.from\\");

require(\\"core-js/modules/es.array.iterator\\");

require(\\"core-js/modules/es.array.slice\\");

require(\\"core-js/modules/es.function.name\\");

require(\\"core-js/modules/es.object.to-string\\");

require(\\"core-js/modules/es.promise\\");

require(\\"core-js/modules/es.regexp.to-string\\");

require(\\"core-js/modules/es.string.iterator\\");

require(\\"core-js/modules/web.dom-collections.iterator\\");

var _asyncToGenerator2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/asyncToGenerator\\"));

var _regenerator = _interopRequireDefault(require(\\"@babel/runtime/regenerator\\"));

require(\\"regenerator-runtime/runtime\\");

function _createForOfIteratorHelper(o) { if (typeof Symbol === \\"undefined\\" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (o = _unsupportedIterableToArray(o))) { var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError(\\"Invalid attempt to iterate non-iterable instance.\\\\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.\\"); } var it, normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it.return != null) it.return(); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === \\"string\\") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === \\"Object\\" && o.constructor) n = o.constructor.name; if (n === \\"Map\\" || n === \\"Set\\") return Array.from(n); if (n === \\"Arguments\\" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

var _marked = /*#__PURE__*/_regenerator.default.mark(generator);

var constVar = 'someConstVar';
var promise = new Promise(function (resolve) {
  setTimeout(function () {
    resolve(\\"Resolving \\".concat(constVar));
  }, 100);
});

function generator() {
  return _regenerator.default.wrap(function generator$(_context) {
    while (1) {
      switch (_context.prev = _context.next) {
        case 0:
          _context.next = 2;
          return 1;

        case 2:
          _context.next = 4;
          return 2;

        case 4:
          _context.next = 6;
          return 3;

        case 6:
        case \\"end\\":
          return _context.stop();
      }
    }
  }, _marked);
}

var object = {
  a: {
    b: 'b'
  },
  b: {
    a: 'a'
  }
};

var init = /*#__PURE__*/function () {
  var _ref = (0, _asyncToGenerator2.default)( /*#__PURE__*/_regenerator.default.mark(function _callee() {
    var _object$a;

    var result, _iterator, _step, a;

    return _regenerator.default.wrap(function _callee$(_context2) {
      while (1) {
        switch (_context2.prev = _context2.next) {
          case 0:
            _context2.next = 2;
            return promise();

          case 2:
            result = _context2.sent;
            _iterator = _createForOfIteratorHelper(generator());

            try {
              for (_iterator.s(); !(_step = _iterator.n()).done;) {
                a = _step.value;
                console.log(a);
              }
            } catch (err) {
              _iterator.e(err);
            } finally {
              _iterator.f();
            }

            console.log(object === null || object === void 0 ? void 0 : (_object$a = object.a) === null || _object$a === void 0 ? void 0 : _object$a.b);
            return _context2.abrupt(\\"return\\", result);

          case 7:
          case \\"end\\":
            return _context2.stop();
        }
      }
    }, _callee);
  }));

  return function init() {
    return _ref.apply(this, arguments);
  };
}();

init().then(function (result) {
  console.log(result);
});"
`;

exports[`Babel preset default transpiles ES6+ code properly with runtimeESModules set to true 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\");

require(\\"core-js/modules/es.symbol\\");

require(\\"core-js/modules/es.symbol.description\\");

require(\\"core-js/modules/es.symbol.iterator\\");

require(\\"core-js/modules/es.array.from\\");

require(\\"core-js/modules/es.array.iterator\\");

require(\\"core-js/modules/es.array.slice\\");

require(\\"core-js/modules/es.function.name\\");

require(\\"core-js/modules/es.object.to-string\\");

require(\\"core-js/modules/es.promise\\");

require(\\"core-js/modules/es.regexp.to-string\\");

require(\\"core-js/modules/es.string.iterator\\");

require(\\"core-js/modules/web.dom-collections.iterator\\");

var _asyncToGenerator2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/esm/asyncToGenerator\\"));

var _regenerator = _interopRequireDefault(require(\\"@babel/runtime/regenerator\\"));

require(\\"regenerator-runtime/runtime\\");

function _createForOfIteratorHelper(o) { if (typeof Symbol === \\"undefined\\" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (o = _unsupportedIterableToArray(o))) { var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError(\\"Invalid attempt to iterate non-iterable instance.\\\\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.\\"); } var it, normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it.return != null) it.return(); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === \\"string\\") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === \\"Object\\" && o.constructor) n = o.constructor.name; if (n === \\"Map\\" || n === \\"Set\\") return Array.from(n); if (n === \\"Arguments\\" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

var _marked = /*#__PURE__*/_regenerator.default.mark(generator);

var constVar = 'someConstVar';
var promise = new Promise(function (resolve) {
  setTimeout(function () {
    resolve(\\"Resolving \\".concat(constVar));
  }, 100);
});

function generator() {
  return _regenerator.default.wrap(function generator$(_context) {
    while (1) {
      switch (_context.prev = _context.next) {
        case 0:
          _context.next = 2;
          return 1;

        case 2:
          _context.next = 4;
          return 2;

        case 4:
          _context.next = 6;
          return 3;

        case 6:
        case \\"end\\":
          return _context.stop();
      }
    }
  }, _marked);
}

var object = {
  a: {
    b: 'b'
  },
  b: {
    a: 'a'
  }
};

var init = /*#__PURE__*/function () {
  var _ref = (0, _asyncToGenerator2.default)( /*#__PURE__*/_regenerator.default.mark(function _callee() {
    var _object$a;

    var result, _iterator, _step, a;

    return _regenerator.default.wrap(function _callee$(_context2) {
      while (1) {
        switch (_context2.prev = _context2.next) {
          case 0:
            _context2.next = 2;
            return promise();

          case 2:
            result = _context2.sent;
            _iterator = _createForOfIteratorHelper(generator());

            try {
              for (_iterator.s(); !(_step = _iterator.n()).done;) {
                a = _step.value;
                console.log(a);
              }
            } catch (err) {
              _iterator.e(err);
            } finally {
              _iterator.f();
            }

            console.log(object === null || object === void 0 ? void 0 : (_object$a = object.a) === null || _object$a === void 0 ? void 0 : _object$a.b);
            return _context2.abrupt(\\"return\\", result);

          case 7:
          case \\"end\\":
            return _context2.stop();
        }
      }
    }, _callee);
  }));

  return function init() {
    return _ref.apply(this, arguments);
  };
}();

init().then(function (result) {
  console.log(result);
});"
`;

exports[`Babel preset default transpiles react code properly 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\");

require(\\"core-js/modules/es.symbol\\");

require(\\"core-js/modules/es.symbol.description\\");

require(\\"core-js/modules/es.object.to-string\\");

require(\\"core-js/modules/es.reflect.construct\\");

require(\\"core-js/modules/es.regexp.to-string\\");

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/classCallCheck\\"));

var _createClass2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/createClass\\"));

var _inherits2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/inherits\\"));

var _possibleConstructorReturn2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/possibleConstructorReturn\\"));

var _getPrototypeOf2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/getPrototypeOf\\"));

var _react = _interopRequireDefault(require(\\"react\\"));

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === \\"undefined\\" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === \\"function\\") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

// eslint-disable-line
var Component = function Component(_ref) {
  var title = _ref.title,
      url = _ref.url,
      description = _ref.description;
  return /*#__PURE__*/_react.default.createElement(\\"div\\", {
    className: \\"component-test-babel\\"
  }, /*#__PURE__*/_react.default.createElement(\\"a\\", {
    href: url,
    title: title
  }, /*#__PURE__*/_react.default.createElement(\\"h1\\", null, title)), /*#__PURE__*/_react.default.createElement(\\"p\\", null, description));
};

Component.defaultProps = {
  title: '',
  url: '',
  description: ''
};

var ClassComponent = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(ClassComponent, _React$Component);

  var _super = _createSuper(ClassComponent);

  function ClassComponent() {
    (0, _classCallCheck2.default)(this, ClassComponent);
    return _super.apply(this, arguments);
  }

  (0, _createClass2.default)(ClassComponent, [{
    key: \\"componentDidMount\\",
    value: function componentDidMount() {// doing something
    }
  }, {
    key: \\"render\\",
    value: function render() {
      var title = this.props.title;
      return /*#__PURE__*/_react.default.createElement(\\"div\\", {
        className: \\"Class Component\\"
      }, /*#__PURE__*/_react.default.createElement(\\"h1\\", null, title));
    }
  }]);
  return ClassComponent;
}(_react.default.Component);

var _default = Component;
exports.default = _default;"
`;

exports[`Babel preset default transpiles without removing proptypes in dev mode 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\");

require(\\"core-js/modules/es.symbol\\");

require(\\"core-js/modules/es.symbol.description\\");

require(\\"core-js/modules/es.object.to-string\\");

require(\\"core-js/modules/es.reflect.construct\\");

require(\\"core-js/modules/es.regexp.to-string\\");

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/classCallCheck\\"));

var _createClass2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/createClass\\"));

var _inherits2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/inherits\\"));

var _possibleConstructorReturn2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/possibleConstructorReturn\\"));

var _getPrototypeOf2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/getPrototypeOf\\"));

var _react = _interopRequireDefault(require(\\"react\\"));

var _propTypes = _interopRequireDefault(require(\\"prop-types\\"));

var _this = void 0,
    _jsxFileName = \\"\\";

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === \\"undefined\\" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === \\"function\\") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

// eslint-disable-line
var Component = function Component(_ref) {
  var title = _ref.title,
      url = _ref.url,
      description = _ref.description;
  return /*#__PURE__*/_react.default.createElement(\\"div\\", {
    className: \\"component-test-babel\\",
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 5,
      columnNumber: 2
    }
  }, /*#__PURE__*/_react.default.createElement(\\"a\\", {
    href: url,
    title: title,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 6,
      columnNumber: 3
    }
  }, /*#__PURE__*/_react.default.createElement(\\"h1\\", {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 7,
      columnNumber: 4
    }
  }, title)), /*#__PURE__*/_react.default.createElement(\\"p\\", {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 9,
      columnNumber: 3
    }
  }, description));
};

Component.propTypes = {
  title: _propTypes.default.string,
  url: _propTypes.default.string,
  description: _propTypes.default.string
};
Component.defaultProps = {
  title: '',
  url: '',
  description: ''
};

var ClassComponent = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(ClassComponent, _React$Component);

  var _super = _createSuper(ClassComponent);

  function ClassComponent() {
    (0, _classCallCheck2.default)(this, ClassComponent);
    return _super.apply(this, arguments);
  }

  (0, _createClass2.default)(ClassComponent, [{
    key: \\"componentDidMount\\",
    value: function componentDidMount() {// doing something
    }
  }, {
    key: \\"render\\",
    value: function render() {
      var title = this.props.title;
      return /*#__PURE__*/_react.default.createElement(\\"div\\", {
        className: \\"Class Component\\",
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 33,
          columnNumber: 4
        }
      }, /*#__PURE__*/_react.default.createElement(\\"h1\\", {
        __self: this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 34,
          columnNumber: 5
        }
      }, title));
    }
  }]);
  return ClassComponent;
}(_react.default.Component);

ClassComponent.propTypes = {
  title: _propTypes.default.string.isRequired
};
var _default = Component;
exports.default = _default;"
`;

exports[`Babel preset default transpiles without transforming es6 imports 1`] = `
"import \\"core-js/modules/es.symbol\\";
import \\"core-js/modules/es.symbol.description\\";
import \\"core-js/modules/es.symbol.iterator\\";
import \\"core-js/modules/es.array.from\\";
import \\"core-js/modules/es.array.iterator\\";
import \\"core-js/modules/es.array.slice\\";
import \\"core-js/modules/es.function.name\\";
import \\"core-js/modules/es.object.to-string\\";
import \\"core-js/modules/es.promise\\";
import \\"core-js/modules/es.regexp.to-string\\";
import \\"core-js/modules/es.string.iterator\\";
import \\"core-js/modules/web.dom-collections.iterator\\";
import _asyncToGenerator from \\"@babel/runtime/helpers/esm/asyncToGenerator\\";
import _regeneratorRuntime from \\"@babel/runtime/regenerator\\";
import \\"regenerator-runtime/runtime\\";

function _createForOfIteratorHelper(o) { if (typeof Symbol === \\"undefined\\" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (o = _unsupportedIterableToArray(o))) { var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError(\\"Invalid attempt to iterate non-iterable instance.\\\\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.\\"); } var it, normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it.return != null) it.return(); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === \\"string\\") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === \\"Object\\" && o.constructor) n = o.constructor.name; if (n === \\"Map\\" || n === \\"Set\\") return Array.from(n); if (n === \\"Arguments\\" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

var _marked = /*#__PURE__*/_regeneratorRuntime.mark(generator);

var constVar = 'someConstVar';
var promise = new Promise(function (resolve) {
  setTimeout(function () {
    resolve(\\"Resolving \\".concat(constVar));
  }, 100);
});

function generator() {
  return _regeneratorRuntime.wrap(function generator$(_context) {
    while (1) {
      switch (_context.prev = _context.next) {
        case 0:
          _context.next = 2;
          return 1;

        case 2:
          _context.next = 4;
          return 2;

        case 4:
          _context.next = 6;
          return 3;

        case 6:
        case \\"end\\":
          return _context.stop();
      }
    }
  }, _marked);
}

var object = {
  a: {
    b: 'b'
  },
  b: {
    a: 'a'
  }
};

var init = /*#__PURE__*/function () {
  var _ref = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime.mark(function _callee() {
    var _object$a;

    var result, _iterator, _step, a;

    return _regeneratorRuntime.wrap(function _callee$(_context2) {
      while (1) {
        switch (_context2.prev = _context2.next) {
          case 0:
            _context2.next = 2;
            return promise();

          case 2:
            result = _context2.sent;
            _iterator = _createForOfIteratorHelper(generator());

            try {
              for (_iterator.s(); !(_step = _iterator.n()).done;) {
                a = _step.value;
                console.log(a);
              }
            } catch (err) {
              _iterator.e(err);
            } finally {
              _iterator.f();
            }

            console.log(object === null || object === void 0 ? void 0 : (_object$a = object.a) === null || _object$a === void 0 ? void 0 : _object$a.b);
            return _context2.abrupt(\\"return\\", result);

          case 7:
          case \\"end\\":
            return _context2.stop();
        }
      }
    }, _callee);
  }));

  return function init() {
    return _ref.apply(this, arguments);
  };
}();

init().then(function (result) {
  console.log(result);
});"
`;

exports[`Babel preset default transpiles wordpress code properly 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\");

require(\\"core-js/modules/es.symbol\\");

require(\\"core-js/modules/es.symbol.description\\");

require(\\"core-js/modules/es.object.to-string\\");

require(\\"core-js/modules/es.reflect.construct\\");

require(\\"core-js/modules/es.regexp.to-string\\");

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _element = require(\\"@wordpress/element\\");

var _classCallCheck2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/classCallCheck\\"));

var _createClass2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/createClass\\"));

var _inherits2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/inherits\\"));

var _possibleConstructorReturn2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/possibleConstructorReturn\\"));

var _getPrototypeOf2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/getPrototypeOf\\"));

var _react = _interopRequireDefault(require(\\"react\\"));

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === \\"undefined\\" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === \\"function\\") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

// eslint-disable-line
var Component = function Component(_ref) {
  var title = _ref.title,
      url = _ref.url,
      description = _ref.description;
  return (0, _element.createElement)(\\"div\\", {
    className: \\"component-test-babel\\"
  }, (0, _element.createElement)(\\"a\\", {
    href: url,
    title: title
  }, (0, _element.createElement)(\\"h1\\", null, title)), (0, _element.createElement)(\\"p\\", null, description));
};

Component.defaultProps = {
  title: '',
  url: '',
  description: ''
};

var ClassComponent = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(ClassComponent, _React$Component);

  var _super = _createSuper(ClassComponent);

  function ClassComponent() {
    (0, _classCallCheck2.default)(this, ClassComponent);
    return _super.apply(this, arguments);
  }

  (0, _createClass2.default)(ClassComponent, [{
    key: \\"componentDidMount\\",
    value: function componentDidMount() {// doing something
    }
  }, {
    key: \\"render\\",
    value: function render() {
      var title = this.props.title;
      return (0, _element.createElement)(\\"div\\", {
        className: \\"Class Component\\"
      }, (0, _element.createElement)(\\"h1\\", null, title));
    }
  }]);
  return ClassComponent;
}(_react.default.Component);

var _default = Component;
exports.default = _default;"
`;
